File: src/main/java/com/bytetrend/sandbox/java/hackerrank/Staircase.java
package com.bytetrend.sandbox.java.hackerrank;

import java.util.Arrays;
import java.util.Scanner;

/**
 * Consider a staircase of size n=4:
 * <p>
 *    #
 *   ##
 *  ###
 * ####
 * Observe that its base and height are both equal to n, and
 * the image is drawn using # symbols and spaces.
 * The last line is not preceded by any spaces.
 * <p>
 * Write a program that prints a staircase of size n.
 * <p>
 * Input Format
 * <p>
 * A single integer, , denoting the size of the staircase.
 */
public class Staircase {
    public static void draw(int n) {
        byte[] blank = new byte[n];
        Arrays.fill(blank, (byte) 0x20);
        for (int i = n - 1; i >= 0; i--) {
            Arrays.fill(blank, i, i + 1, (byte) '#');
            System.out.println(new String(blank));
        }
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        draw(n);
    }
}

File: src/main/scala/com/bytetrend/sandbox/scala/hackerrank/Staircase.scala
package com.bytetrend.sandbox.scala.hackerrank


object Staircase {

  /**
    * https://www.hackerrank.com/challenges/staircase
    * Consider a staircase of size n = 4:
    *
    *    #
    *   ##
    *  ###
    * ####
    * Observe that its base and height are both equal to n, and the image is drawn using # symbols and spaces. The last line is not preceded by any spaces.
    * Write a program that prints a staircase of size n.
    * *
    * Input Format
    * *
    * A single integer, n, denoting the size of the staircase.
    * *
    * Output Format
    * *
    * Print a staircase of size n using # symbols and spaces.
    * *
    * Note: The last line must have 0 spaces in it.
    * *
    * Sample Input
    * *
    * 6
    *
    * @param n
    * @return
    */
  def staircase(n: Int): Seq[String] = {
    for {
      x <- 1 to n
    } yield " " * (n - x) + "#" * x
  }

  def main(args: Array[String]) {
    val sc = new java.util.Scanner(System.in);
    val n = sc.nextInt();
    println(staircase(n).mkString("\n"))
  }

}